src/sage/groups/abelian_gps/abelian_group.py

```diff
@@ -1708,16 +1712,19 @@ def __init__(self, ambient, gens, names="f", category=None):
         H = libgap(ambient).Subgroup(H_gens)
 
         invs = H.TorsionSubgroup().AbelianInvariants().sage()
+        gens_orders = [order_sage for g in H.GeneratorsOfGroup() if (order_sage := g.Order().sage()) is not infinity]
+
         rank = len([1 for g in H.GeneratorsOfGroup()
                     if g.Order().sage() is infinity])
         invs += [0] * rank
-
+        gens_orders += [0] * rank
         self._abinvs = invs
         invs = tuple(ZZ(i) for i in invs)
+        gens_orders = tuple(ZZ(i) for i in gens_orders)
```

Comment:
```suggestion
```
This has become unnecessary as the above suggestion should be faster (less transient objects).